code/modules/mob/living/carbon/alien/special/facehugger.dm

```diff
@@ -413,11 +416,170 @@
+	icon_state = "headcrab"
+	item_state = "headcrab"
+	body_parts_covered = HEAD
+	slot_flags = SLOT_HEAD
+	clothing_flags = null
+	canremove = 0  //You need to resist out of it.
+	cant_remove_msg = "The headcrab is latched on tight!"
+
+/obj/item/clothing/mask/facehugger/headcrab/attack_hand(mob/user)
+	var/mob/living/carbon/human/target = user
+	if(target && target.head == src)
+		target.resist()
+	else
+		..()
+
+/obj/item/clothing/mask/facehugger/headcrab/followtarget()
+	if(!real)
+		return // Why are you trying to path stupid toy
+	if(!target)
+		findtarget()
+		return
+	if(loc && isturf(loc) && !attached && !stat && nextwalk <= world.time)
+		nextwalk = world.time + walk_speed
+		var/dist = get_dist(loc, target.loc)
+		if(dist > 5)
+			target = null
+			return //We'll let the facehugger do nothing for a bit, since it's fucking up.
+
+		var/obj/item/clothing/mask/facehugger/headcrab/F = target.is_wearing_item(/obj/item/clothing/mask/facehugger/headcrab, slot_head)
+		if(F && !F.sterile) // Toys won't prevent real huggers
+			findtarget()
+			return
+		else
+			playsound(src, pick('sound/items/headcrab_attack1.ogg', 'sound/items/headcrab_attack2.ogg', 'sound/items/headcrab_attack3.ogg'), 50, 0)
+			throw_at(target, 3, 1)
+			if(dist == 0)	
+				if(CanHug(target, src) && isturf(target.loc)) //Fix for hugging through mechs and closets
+					Attach(target)
+					return
+				else
+					walk(src,0)		
+					findtarget()
+					return
+
+
+/obj/item/clothing/mask/facehugger/headcrab/Attach(mob/living/L)
+	if(isalien(L))
+		return FALSE
+	if(attached)
+		return FALSE
+	if(!Adjacent(L))
+		return FALSE
+	if(loc == L)
+		return FALSE
+	if(stat != CONSCIOUS)
+		return FALSE
+	if(!sterile)
+		L.take_organ_damage(strength, 0) //done here so that even borgs and humans in helmets take damage
+
+	L.visible_message("<span class='danger'>\The [src] leaps at [L]'s face!</span>")
+
+	if(!CanHug(L, src))
+		return FALSE
+
+	if(ishuman(L))
+		var/mob/living/carbon/human/H = L
+		var/obj/item/head_protection = H.get_body_part_coverage(HEAD)
+		if(!real && head_protection)
+			return //Toys really shouldn't be forcefully removing gear
+
+		if(head_protection)
+			var/rng = 80	//80% chance to remove the hat if it isn't a rig
+			if(istype(head_protection, /obj/item/clothing/head/helmet/space/rig))
+				rng = CHANCE_TO_REMOVE_SPECIAL_HEADWEAR
+			if(prob(rng)) // Temporary balance change, all mouth-covering hats will be more effective
+				H.visible_message("<span class='danger'>\The [src] smashes against [H]'s \the [head_protection], and rips it off in the process!</span>")
+				H.drop_from_inventory(head_protection)
+				GoIdle(TIME_IDLE_AFTER_HEAD_DENIED)
+				return
+			else
+				H.visible_message("<span class='danger'>\The [src] bounces off of \the [head_protection]!</span>")
+				if(prob(CHANCE_TO_DIE_AFTER_HEAD_DENIED) && !sterile)
+					death()
+					return
+				else
+					GoIdle(TIME_IDLE_AFTER_HEAD_DENIED)
+					return
+
+	if(iscarbon(L))
```

Comment:
I don't think this is a good idea, there's lots of kinds of carbons and it'll be pretty janky. Best to restrict this to humans for now